Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -43,20 +26,50 @@ public void updateQuality() {
             if (items[i].sellIn < 0) {
                 if (!items[i].name.equals("Aged Brie")) {
                     if (!items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].quality > 0) {
-                            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                                items[i].quality = items[i].quality - 1;
-                            }
-                        }
+                        Sulfuras(items);
                     } else {
                         items[i].quality = items[i].quality - items[i].quality;
                     }
                 } else {
-                    if (items[i].quality < 50) {
-                        items[i].quality = items[i].quality + 1;
+                    Aged(items);
+                }
+            }
+        }
+    }
+
+    public String Sulfuras(items) {
+        for (int i = 0; i < items.length; i++) {
```

Comment:
como primeira iteración está ben, reparte lóxica por item  pero necesita traballo para reducir a súa complexidade e simplicar of ifs e os niveis de indentación